examples/plot_poisson_hmm.py

```diff
@@ -0,0 +1,114 @@
+ax.set_xticks(range(0, earthquakes.size, 10))
+ax.set_xticklabels(range(1906, 2007, 10))
+ax.set_xlabel('Year')
+ax.set_ylabel('Count')
+fig.show()
+
+# %%
+# Now, fit a Poisson Hidden Markov Model to the data.
+
+scores = list()
+models = list()
+for n_components in range(1, 5):
+    for idx in range(50):
+        # define our hidden Markov model
+        model = hmm.PoissonHMM(n_components=3, random_state=idx)
+        model.fit(earthquakes[:, None])
+        models.append(model)
+        scores.append(model.score(earthquakes[:, None]))
+        print(f'Converged: {model.monitor_.converged}\t\t'
+              f'Score: {scores[-1]}')
+
+# get the best model
+model = models[np.argmax(scores)]
+print(f'The best model had a score of {max(scores)} and '
+      f'{model.n_components} components')
+
+# use the Viterbi algorithm to predict the most likely sequence of states
+# given the model
+states = model.predict(earthquakes[:, None])
+
+# %%
+# Let's plot the waiting times from our most likely series of states of
+# earthquake activity with the earthquake data. As we can see, the
+# model with the maximum likelihood had three states which may reflect
+# times of low, medium and high earthquake danger.
+
+# plot model states over time
+fig, ax = plt.subplots()
+ax.plot(model.lambdas_[states], ".-", ms=6, mfc="orange")
+ax.plot(earthquakes)
+ax.set_title('States compared to generated')
+ax.set_xlabel('State')
+fig.show()
+
+# %%
+# Fortunately, 2006 ended with a period of relative tectonic stability, and,
+# if we look at our transition matrix, we can see that the off-diagonal terms
+# are small, meaning that the state transitions are rare and it's unlikely that
+# there will be high earthquake danger in the near future. The transitions from
+# state 1, which has longest waiting time between earthquakes to state 2, which
+# has the shortest is 0.001, meaning that the years following 2006 are unlikely
+# to have more earthquakes than average.
+
+fig, ax = plt.subplots()
+ax.imshow(model.transmat_, aspect='auto', cmap='spring')
+ax.set_title('Transition Matrix')
+ax.set_xlabel('State To')
+ax.set_ylabel('State From')
+fig.show()
+
+# %%
+# Finally, let's look at the distribution of earthquakes compared to our
+# waiting time parameter values. We can see that our model fits the
+# distribution fairly well, replicating results from the reference.
+
+# get probabilities for each state given the data, take the average
+# to find the proportion of time in that state
+prop_per_state = model.predict_proba(earthquakes[:, None]).mean(axis=0)
+
+# earthquake counts to plot
+bins = sorted(np.unique(earthquakes))
+
+fig, ax = plt.subplots()
+ax.hist(earthquakes, bins=bins, density=True)
+ax.plot(bins, np.dot(poisson.pmf(bins, model.lambdas_).T,
+                     prop_per_state[:, None]))
+ax.set_title('Histogram of Earthquakes with Fitted Poisson States')
+ax.set_xlabel('Number of Earthquakes')
+ax.set_ylabel('Proportion')
+fig.show()
```

Comment:
You can drop the fig.show(), and you need a plt.show() at the end so that the figures don't get hidden if running straight from the terminal (`python examples/plot_poisson_hmm.py`).